lib/marked.js

```diff
@@ -363,10 +365,20 @@ Lexer.prototype.token = function(src, top) {
           if (!loose) loose = next;
         }
 
+        // Check for task list items
+        istask = /^\[[ xX]\]/.test(item);
```

Comment:
a space seems to be required after `[ ]` other wise it is rendered like text

with space:

- [x] test

without space:

- [x]test